numpy/_core/src/multiarray/stringdtype/casts.cpp

```diff
@@ -19,36 +23,243 @@
 #include "dtype.h"
 #include "utf8_utils.h"
 
+#include "casts.h"
+
+
+// Get the shortname letter associated with the given type number.
+NPY_TYPECHAR typenum_to_typechar(NPY_TYPES typenum) {
```

Comment:
AFAICS this is unused and can be deleted

Reply:
Yes, this is left over from some now unused error messaging :+1: